internal/reloader/handler_race_test.go

```diff
@@ -0,0 +1,64 @@
+//go:build race
+// +build race
+
+package reloader
+
+import (
+	"context"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAtomicHandlerRaces(t *testing.T) {
```

Comment:
If you explicitly say that we use `atomic.Value` then this test is not necessary I suppose. Wrapping `atomic.Value` in another type does make it "necessary" but even then I'd just minimize code written... this is testing the Go standard library in a way.

Reply:
I wrote a test to make sure that I adhered to the coverage guidelines given first, second to ensure future guarantees that this code remained correct as it ages. I am happy to remove it if you would like.

Reply:
Up to you!